src/RageSoundMixBuffer.cpp

```diff
@@ -14,14 +14,27 @@ static bool g_bVector = Vector::CheckForVector();
 
 RageSoundMixBuffer::RageSoundMixBuffer()
 {
-	m_iBufSize = m_iBufUsed = 0;
-	m_pMixbuf = nullptr;
+	// Set m_iBufSize to 2MB.
+	m_iBufSize = 2 * 1024 * 1024 / sizeof(float);
+
+	// Allocate memory
+	m_pMixbuf = static_cast<float*>(std::malloc(m_iBufSize * sizeof(float)));
+
+	// Check if memory allocation was successful
+	if (m_pMixbuf == nullptr)
+	{
+		std::exit(EXIT_FAILURE);
+	}
+
+	memset(m_pMixbuf, 0, m_iBufSize * sizeof(float));
```

Comment:
I think we can convert to std::memset here since we're changing the others anyways.